src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/CpsPackageReferenceProject.cs

```diff
@@ -358,10 +358,48 @@ private async Task SetPackagePropertyValueAsync(IProjectProperties metadata, str
                 propertyValue);
         }
 
-        public override async Task<bool> UninstallPackageAsync(PackageIdentity packageIdentity, INuGetProjectContext nuGetProjectContext, CancellationToken token)
+        public override Task<bool> UninstallPackageAsync(PackageIdentity packageIdentity, INuGetProjectContext nuGetProjectContext, CancellationToken token)
         {
-            var configuredProject = await _unconfiguredProject.GetSuggestedConfiguredProjectAsync();
-            await configuredProject?.Services.PackageReferences.RemoveAsync(packageIdentity.Id);
+            throw new InvalidOperationException("For CPSPackageReferenceProject, the uninstall method that can handle conditional targeting must be called instead.");
+        }
+
+        public override async Task<bool> UninstallPackageAsync(string packageId, BuildIntegratedInstallationContext installationContext, CancellationToken token)
+        {
+            token.ThrowIfCancellationRequested();
+            if (packageId == null) throw new ArgumentNullException(nameof(packageId));
```

Comment:
empty packageId are okay?

Reply:
No, I just didn't check for it. 

I can add that check too.